pyckel/src/lib.rs

```diff
@@ -1,62 +1,26 @@
 use std::io::Cursor;
 
-use nickel_lang::{
-    error::{Error, SerializationError},
-    eval::cache::CBNCache,
-    program::Program,
-    serialize,
-};
+use nickel_lang::{eval::cache::CBNCache, program::Program, serialize};
 
 use pyo3::{create_exception, exceptions::PyException, prelude::*};
 
 create_exception!(pyckel, NickelException, PyException);
 
-// see https://pyo3.rs/v0.17.3/function/error_handling.html#foreign-rust-error-types
-struct NickelError(Error);
-struct NickelSerializationError(SerializationError);
-
-impl From<Error> for NickelError {
-    fn from(value: Error) -> Self {
-        Self(value)
-    }
-}
-
-impl std::convert::From<NickelError> for PyErr {
-    fn from(err: NickelError) -> PyErr {
-        match err {
-            // TODO better exceptions
-            NickelError(error) => NickelException::new_err(format!("{error:?}")),
-        }
-    }
-}
-
-impl From<SerializationError> for NickelSerializationError {
-    fn from(value: SerializationError) -> Self {
-        Self(value)
-    }
-}
-
-impl std::convert::From<NickelSerializationError> for PyErr {
-    fn from(err: NickelSerializationError) -> PyErr {
-        match err {
-            // TODO better exceptions
-            NickelSerializationError(error) => NickelException::new_err(format!("{error:?}")),
-        }
-    }
-}
-
 /// Evaluate from a Python str of a Nickel expression to a Python str of the resulting JSON.
 #[pyfunction]
 pub fn run(s: String) -> PyResult<String> {
     let mut program: Program<CBNCache> =
         Program::new_from_source(Cursor::new(s.to_string()), "python")?;
 
-    let term = program.eval_full().map_err(NickelError)?;
+    let term = program
+        .eval_full()
+        .map_err(|err| NickelException::new_err(program.report_as_str(err)))?;
```

Comment:
That's probably my own proposal to inline this expression :sweat_smile: , but I didn't realize we use it 3 times. Maybe it's better to factor it out in a dedicated function, in the end?

Reply:
Fixed in 54e7cad3

Funny thing is, it did not make it shorter. 😅